UI/Components/EditorElement/EditorElement.xaml.cs

```diff
@@ -568,39 +546,35 @@ private void Editor_MouseDown(object sender, MouseButtonEventArgs e)
         private void TextArea_PreviewKeyDown(object sender, KeyEventArgs e)
         {
             e.Handled = ISAC_EvaluateKeyDownEvent(e.Key);
-            if (!e.Handled
-            ) //one could ask why some key-bindings are handled here. Its because spedit sends handled flags for ups&downs and they are therefore not able to processed by the central code.
+            if (!e.IsDown || e.Handled)
             {
-                if (e.KeyboardDevice.IsKeyDown(Key.LeftCtrl))
-                {
-                    if (e.KeyboardDevice.IsKeyDown(Key.LeftAlt))
-                    {
-                        if (e.Key == Key.Down)
-                        {
-                            DuplicateLine(true);
-                            e.Handled = true;
-                        }
-                        else if (e.Key == Key.Up)
-                        {
-                            DuplicateLine(false);
-                            e.Handled = true;
-                        }
-                    }
-                    else
-                    {
-                        if (e.Key == Key.Down)
-                        {
-                            MoveLine(true);
-                            e.Handled = true;
-                        }
-                        else if (e.Key == Key.Up)
-                        {
-                            MoveLine(false);
-                            e.Handled = true;
-                        }
-                    }
-                }
+                return;
+            }
+
+            var key = e.Key;
+            var modifiers = Keyboard.Modifiers;
+
+            if (key == Key.System)
+            {
+                key = e.SystemKey;
+            }
+
+            if (key == Key.LeftCtrl ||
+                key == Key.RightCtrl ||
+                key == Key.LeftAlt ||
+                key == Key.RightAlt ||
+                key == Key.LeftShift ||
+                key == Key.RightShift ||
+                key == Key.LWin ||
+                key == Key.RWin ||
+                key == Key.Clear ||
+                key == Key.OemClear ||
+                key == Key.Apps)
+            {
+                return;
             }
```

Comment:
Switch case can look better here? (instead of using `key == X` you can do `case X:` for each one)